src/wrapt/wrapt.pyi

```diff
@@ -0,0 +1,13 @@
+from typing import Any, Generic, TypeVar, Callable, Optional
+
+F = TypeVar('F', bound=Callable[..., Any])
+A = TypeVar('A', bound=Callable[..., Any])
+T = TypeVar("T", bound=Any)
+
+def decorator(wrapper: F, enabled: Optional[bool] = None, adapter: Optional[A] = None) -> F: ...
```

Comment:
Due to https://github.com/microsoft/pyright/issues/5929 the re-export in `__init__.py` needs to be updated too.

For example, `from .decorators import decorator as decorator`

The issue affects `pyright` but not `mypy`.

You can test this by e.g.

```command
poetry add git+https://github.com/GrahamDumpleton/wrapt.git#a2dddf6e10536f4cd4bfc96ff1a706cd71b0694c
poetry add pyright
poetry add mypy

pyright some_test_file.py
mypy some_test_file.py
```

Reply:
An alternative to `from ... import A as A` would be to provide explicit `__all__` in `wrapt/__init__.py`.